src/component/page-list/page-tile-container.tsx

```diff
@@ -0,0 +1,94 @@
+import * as MobX from 'mobx';
+import { observer } from 'mobx-react';
+import { PageRef } from '../../store/page/page-ref';
+import { PageTileComposite } from './page-tile-composite';
+import * as React from 'react';
+import { Store } from '../../store/store';
+
+export interface PageTileContainerProps {
+	focused: boolean;
+	onClick: React.MouseEventHandler<HTMLElement>;
+	page: PageRef;
+}
+
+@observer
+export class PageTileContainer extends React.Component<PageTileContainerProps> {
+	@MobX.observable public editable: boolean = false;
+	@MobX.observable public inputValue: string = '';
+
+	public constructor(props: PageTileContainerProps) {
+		super(props);
+		this.inputValue = this.inputValue ? this.inputValue : this.props.page.getName();
+
+		this.handleChange = this.handleChange.bind(this);
+		this.handleClick = this.handleClick.bind(this);
+		this.handleDoubleClick = this.handleDoubleClick.bind(this);
+		this.handleKeyDown = this.handleKeyDown.bind(this);
+	}
+
+	@MobX.action
+	protected handleBlur(): void {
+		this.editable = false;
+	}
+
+	@MobX.action
+	protected handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
+		this.inputValue = e.target.value;
+	}
+
+	@MobX.action
+	protected handleClick(e: React.MouseEvent<HTMLElement>): void {
+		this.props.onClick(e);
+
+		setTimeout(() => {
```

Comment:
Timeout solutions are almost always dirty and incorrect ;-)

Reply:
@TheReincarnator you are right and it's gone